lib/synapse/retry_policy.rb

```diff
@@ -0,0 +1,45 @@
+module Synapse
+  module RetryPolicy
+    def with_retry(options = {}, &callback)
+      max_attempts = options['max_attempts'] || 1
+      max_delay = options['max_delay'] || Float::INFINITY
```

Comment:
if not set, we are going to have infinite delay? which means we will hang on here?

Reply:
change to 3600 seconds